### PR TITLE
Add a broadcast client(s)/server example.

### PR DESCRIPTION
**Pull request by HazenBabcock**
This pull request adds a broadcast server (and clients) example as well as the documentation as described in issue #653 .

### CODE REVIEW

File: example/broadcast_server.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import asyncio
+import time
+import websockets
+
+async def broadcast():
+    while True:
+        message = str(time.time())
+        await asyncio.gather(
+            *[ws.send(message) for ws in CLIENTS],
+            return_exceptions=False,
```
Comment (hilbix) [inline example/broadcast_server.py]: Haven't tested this, but IMHO there should be `return_exceptions=True`, as from the docs: "exceptions are ... aggregated in the result list".  (We probably do not want to see `ConnectionClosed` exceptions propagated into `broadcast()`.)

File: example/broadcast_server.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python
+import asyncio
+import time
+import websockets
+
+async def broadcast():
+    while True:
+        message = str(time.time())
+        await asyncio.gather(
```
Comment (hilbix) [inline example/broadcast_server.py]: `await`?  This makes `broadcast()` sleep until `asyncio.gather()` is fullfilled!  Haven't testet this, but IMHO if one client hangs, so `ws.send()` takes ages to finish, `broadcast()` will sit there and wait.  Perhaps forever.  This might be wanted .. or rather not.  I am still learning AsyncIO but this exactly is the most common pattern which most commonly annoys me most:  How to properly express a fire-and-forget-pattern in AsyncIO which is expressed in bash as trivial as `( something & )`?!?  It must be done safely, as else GC might finalize the objects or we have some memory leak!  The entry point probably is `task = asyncio.ensure_future(asyncio.gather(...` (to be Python 3.6 compatible).  But this is only a third of the solution.  It now leaves us (only me?) with a dangling `task` object which now must be stuffed somewhere safe to keep it away from GC. And when the task is finished, this stuffing must be taken back to free memory again, perhaps with some PITA like `task.add_done_callback()`.  (Didn't I just not come across the correct pattern in AsyncIO for this yet, or is it really so hard to express in AsyncIO?)

File: example/broadcast_server.py (hunk 1 above)
Comment (HazenBabcock) [reply example/broadcast_server.py]: Not sure, I think it is this way so that exceptions immediately propagate into `handler` which removes the corresponding websocket from the `CLIENTS` set. However it has been a while and my understanding of `asnycio` is poor. My testing (as I recall) consisted of running several clients and verifying that if I killed one with CTRL-C everything else kept running as expected.

File: example/broadcast_server.py (hunk 2 above)
Comment (hilbix) [reply example/broadcast_server.py]: OK, forget this.  Just remove the `await` and it should work, as according to my tests (Ubuntu 18.04) one can call `asyncio.gather()` to actually run the awaitables.

Also the tests shows, that `asyncio.ensure_future(g)` just returns `g` when `g = asyncio.gather()`.  Hence this is not changing anything.

I am still a bit puzzled, though, as I cannot see such a run-and-forget guarantee in the `asyncio` sources, **so this behavior might be accidental**.  And nowhere in the docs I found any guarantee, that calling `asyncio.gather()` instead of `await asyncio.gather()` is the correct way to run things in the background.

The only thing I found in the code, which is certainly not accidental, is that following pattern indeed runs a coroutine in background, safely:

    asyncio.Task(coro())

So there is no need to keep the task object around nor to clean things up afterwards.  From the source of `class Task`:

https://github.com/python/asyncio/blob/a791d884b9110b59bada18c2dddd399e6adc40fd/asyncio/tasks.py#L69-L78

```
    def __init__(self, coro, *, loop=None):
        assert coroutines.iscoroutine(coro), repr(coro)
        super().__init__(loop=loop)
        if self._source_traceback:
            del self._source_traceback[-1]
        self._coro = coro
        self._fut_waiter = None
        self._must_cancel = False
        self._loop.call_soon(self._step)
        self.__class__._all_tasks.add(self)
```

This `self.__class__._all_tasks.add(self)` keeps the Task-object alive even if you throw it away afterwards.

**Why can't I find this essential detail prominently presented in the Docs anywhere?**

TL;DR

As `asyncio.Task()` needs a coroutine, but `asyncio.gather()` does not return one, the probably right solution is just to leave away the `await`

**But I am not sure.**

Comment (HazenBabcock) [reply example/broadcast_server.py]: I believe that you are correct that using `await` makes `broadcast()` wait until `asyncio.gather()` finishes. This might not actually lead to a hang if there is additional buffering occurring in `WebSocketCommonProtocol.send()` or at a lower level in the OS. Eventually you'd hit `write_limit` and fail either way?

Anyway, happy to fix the example if you prefer it without the `await`.

Comment (hilbix) [reply example/broadcast_server.py]: I do not have any powers here (and I do not want any) ;)  My intention was to help, because for me, a pure user, this technically looks as a useful example.  For me it actually was very useful already, because it made me look into the asyncio code and learn a bit better about it.  So when the `await` really is removed, there should be a good comment about why it is missing there.  **But I cannot decide if `await` shall be removed or not.**  And as I am not completely sure, others must decide anyway.

**Comment by aaugustin** [comment]
Thanks for your contribution! @hilbix had a good point; this led to a very long discussion in #653, which I summarized in #986. So this PR is essentially superseded by #986.
